Add remaining deprecation guides for 3.27 (#855)
ember-learn/deprecation-app: content/ember/v3/argument-less-helper-paren-less-invocation.md

---
id: argument-less-helper-paren-less-invocation
title: Invoking Helpers Without Arguments and Parentheses In Named Argument Positions
until: '4.0.0'
since: '3.27'
---

With [contextual helpers](https://emberjs.github.io/rfcs/0432-contextual-helpers.html)

> arriving in Ember, helpers, modifiers and components can increasing be thought

Reviewer: ```suggestion
arriving in Ember, helpers, modifiers and components can increasingly be thought
```

of as first-class variables that can be passed around.

Invoking a helper without arguments or parentheses in named argument positions
can be ambigious and conflicts with this mental model:

```hbs
<SomeComponent @arg={{someHelper}} />
```

In this case, it's ambigious between passing `someHelper` as a value to the

> component to be invoked later, or invoking the helper with no arguments and

Reviewer: ```suggestion
component to be invoked later or invoking the helper with no arguments and
```

passing the result into the component.

> The current behavior is to invoke the helper with no arguments and passing the

Reviewer: ```suggestion
The current behavior is to invoke the helper with no arguments and pass in the
```
